src/internal_rep.rs

```diff
@@ -80,6 +98,25 @@ impl TypeInfo {
             declaration_file: Some(file.clone()), // TODO: Turn into reference
             annotations: get_type_annotations(file, &td.annotations)?,
             decl: Some(td),
+            bound_type: BoundTypeInfo::Unbound,
+        })
+    }
+
+    pub fn new_bound_type(
+        name: HLLString,
+        variant: &str,
+        file: &SimpleFile<String, String>,
+        bound_type: BoundTypeInfo,
+    ) -> Result<TypeInfo, HLLErrors> {
+        Ok(TypeInfo {
+            name: name,
+            inherits: vec![variant.into()], // Does this need to somehow grab the bound parents? Does this work for the single case?
+            is_virtual: true,               // Maybe?
+            list_coercion: false,           // TODO
```

Comment:
Cleaned up all the TODOs here except the decl, thanks.  I *think* None should be an acceptable value for Decl at least for now, although maybe in the long term we want to switch this into an enum instead of an option so that decl can be other things that a TypeDecl (for example a LetBinding).  I don't really see a reason this needs to be a priority right now though.